### ISSUE TITLE
After install.sh, Tails seems to hang before loading the desktop

### ISSUE THREAD

**Issue by garrettr** [freedomofpress/securedrop#1299]
Recently, a SecureDrop user reported:

> However, Tails seemed to hang while it was executing some tasks in the background, just before loading the desktop items - don't know if that makes sense.

I also noticed this recently while testing the Tails auto-configuration, and did some investigation. 

The hang appears to be an unintentional side effect caused by the install script enabling "Additional software packages", an experimental Tails feature. The point of enabling this feature is to auto-install Ansible (a package that does not come with Tails by default) on boot - otherwise, the admin would have to apt-get install ansible at the start of each session.

The problem is due to the way Tails implemented this feature. The package itself is saved via Tails persistence, but Tails always checks to see if an upgrade version of the package is available before installing the cached copy. This sort of makes sense as a security feature (avoids auto-installing a potentially vulnerable custom package), but I think it could be implemented better. The way it is now, loading the desktop is blocked until all of the "additional software packages" are installed, which means waiting for an apt-get update and potentially one or more apt-get upgrades as well. As you have probably noticed, apt-get update is quite slow over Tor. I am reasonably sure that this is the cause of the apparent hang just before loading the desktop items.

---

_Process note:_ **support-request** is a new label I've created to tag GitHub issues derived from issues reported on the new private Redmine server for SecureDrop users.

**Comment by conorsch** [comment]
See also #898.

**Comment by garrettr** [comment]
IIRC, when we were doing release QA testing for 0.3.7, this seemed to have been addressed as part of the "upgrade the Tails auto-configuration to work in Tails 2.x", which was primarily tracked in #1309.

This seemed to be an inherent issue in the implementation of the "Additional software packages" persistence feature, and therefore something that should be fixed in Tails. Perhaps it was between 1.x and 2.x. I tried to confirm this, but unfortunately the [Tails issue tracker](https://labs.riseup.net/code/projects/tails) is inaccessible at the moment because their TLS certificate expired on July 2nd and their site uses HSTS 😞 .

I think we should test this again, to verify that is has indeed been resolved, before closing it. However, I am going to remove it from the 0.3.7 milestone because that release already shipped.
